id_handling.c

```diff
@@ -119,62 +113,94 @@ esp_err_t ubirch_id_context_add(const char* short_name) {
+        ret = ESP_FAIL;
+    }
+    if (kv_delete((char*)short_name, KVKEY_PREVIOUS_SIGNATURE) != ESP_OK) {
+        ESP_LOGE(TAG, "failed to delete previous signature from kv-storage");
+        ret = ESP_FAIL;
+    }
+
+    if (strcmp(short_name, current_id_context.short_name) == 0) {
+        current_id_context.memory_state.raw = 0x00;
+        current_id_context.memory_state.valid = 0;
+        current_id_context.memory_state.id_blob_updated = 0;
+        current_id_context.memory_state.previous_signature_updated = 0;
+    }
+
+    return ret;
+}
 
 const char* ubirch_id_context_get(void) {
     return current_id_context.short_name;
 }
 
 esp_err_t ubirch_id_context_load(const char* short_name) {
-    // FIXME: open region only once and load all key-value pairs
-    uint8_t* pmemory_state_addr = &current_id_context.state;
-    esp_err_t ret = load_value(short_name, KVKEY_PERSISTENT_STATE,
-            (void**)&pmemory_state_addr,
-            sizeof(current_id_context.state));
     ESP_LOGD(TAG, "load: short_name = %s, state = %d", short_name, current_id_context.state);
+    // load id blob
+    unsigned char id_blob_buffer[ID_BLOB_LENGTH];
+    unsigned char* addr = id_blob_buffer;
+    esp_err_t ret = load_value(short_name, KVKEY_ID_BLOB, (void**)&addr, ID_BLOB_LENGTH);
     if (ret != ESP_OK) {
-        current_id_context.memory_state.valid = 0;
-        return ret;
-    }
-    unsigned char* addr = UUID;
-    if ((ret = load_value(short_name, KVKEY_UUID,
-                    (void**)&addr, sizeof(UUID))) != ESP_OK) {
-        current_id_context.memory_state.valid = 0;
         return ret;
     }
-    addr = (unsigned char*)current_password; // FIXME: uuaarrggh
-    if (load_value(short_name, KVKEY_PASSWORD,
-                    (void**)&addr, sizeof(current_password)) != ESP_OK) {
-        current_id_context.state &= ~UBIRCH_ID_STATE_PASSWORD_SET;
-    }
-    addr = ed25519_public_key;
-    if (load_value(short_name, KVKEY_PUBLIC_KEY,
-                    (void**)&addr, sizeof(ed25519_public_key)) != ESP_OK) {
-        current_id_context.state &= ~UBIRCH_ID_STATE_KEYS_CREATED;
-    }
-    addr = ed25519_secret_key;
-    if (load_value(short_name, KVKEY_SECRET_KEY,
-                    (void**)&addr, sizeof(ed25519_secret_key)) != ESP_OK) {
-        current_id_context.state &= ~UBIRCH_ID_STATE_KEYS_CREATED;
-    }
-    time_t* next_key_update_addr = &next_key_update;
-    if (load_value(short_name, KVKEY_NEXT_KEY_UPDATE,
-                    (void**)&next_key_update_addr, sizeof(time_t)) != ESP_OK) {
-        current_id_context.state &= ~UBIRCH_ID_STATE_KEYS_REGISTERED;
-    }
+
+    unsigned char* id_blob_buffer_ptr = id_blob_buffer;
+    // copy state from buffer
+    memcpy(&current_id_context.state, id_blob_buffer_ptr, sizeof(current_id_context.state));
+    id_blob_buffer_ptr = id_blob_buffer_ptr + sizeof(current_id_context.state);
+    ESP_LOGD(TAG, "load: short_name = %s, state = %d", short_name, current_id_context.state);
```

Comment:
copy from ln 159, without new information. can be removed

Reply:
removed